File: examples/python/standby.py
"""
Standby — put the display into low-power standby and wake it again.

The display blanks the panel and stops updating while in standby.
Call exit_standby() to resume normal operation.

Run with a Glider connected to a device running firmware v??? or later.

    python examples/python/standby.py
"""

import time
from glider_api import Display, DisplayConfig

STANDBY_SECS = 3

config = DisplayConfig.glider_standard()

try:
    display = Display.new_with_config(config)
except TypeError as e:
    print(f"Could not connect: {e}")
    raise SystemExit(1)

print("Entering standby...")
display.enter_standby()
print(f"Display is in standby. Waiting {STANDBY_SECS} seconds...")

time.sleep(STANDBY_SECS)

print("Exiting standby...")
display.exit_standby()
print("Done — display resumed.")

File: examples/rust/standby.rs
//! Standby — put the display into low-power standby and wake it again.
//!
//! The display blanks the panel and stops updating while in standby.
//! Call `exit_standby` to resume normal operation.
//!
//! ```
//! cargo run --example standby
//! ```

use std::{thread, time::Duration};

use glider_api::{Display, DisplayConfig};

const STANDBY_SECS: u64 = 3;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let config = DisplayConfig::glider_standard();

    let display = Display::new_with_config(&config).map_err(|e| {
        eprintln!("Could not connect: {e}");
        e
    })?;

    println!("Entering standby...");
    display.enter_standby()?;
    println!("Display is in standby. Waiting {STANDBY_SECS} seconds...");

    thread::sleep(Duration::from_secs(STANDBY_SECS));

    println!("Exiting standby...");
    display.exit_standby()?;
    println!("Done — display resumed.");

    Ok(())
}
